graph/patterns/evolvable-enums.md

```diff
@@ -0,0 +1,302 @@
+| ```enumProperty gt newValue``` | ```400 Bad Request``` | Return entities where enumProperty has a value greater than ```newValue``` |
+| ```enumProperty lt newValue``` | ```400 Bad Request``` | Return entities where enumProperty has a value less than ```newValue``` |
+
+If an evolvable enum is included in an ```$orderby``` clause the actual numeric value of the member should be used to order the collection, after sorting the member should then be replaced with ```unknownFutureValue``` when the ```Preference: include-unknown-enum-members``` header is absent.
+
+## Examples
+
+---
+
+For the following examples we will consider the ```managedDevice``` entity which refers to the ```managedDeviceArchitecture``` enum type.
+
+```xml
+<!-- Simplified entity for example purposes -->
+<EntityType Name="managedDevice" BaseType="graph.entity">
+    <Property Name="displayName" Type="Edm.String" />
+    <Property Name="processorArchitecture" Type="graph.managedDeviceArchitecture"/>
+</EntityType>
+```
+
+When the ```managedDeviceArchitecture``` enum was initially published to Graph it was defined as below:
+
+```xml
+<!-- Slightly modified enum for example purposes -->
+<EnumType Name="managedDeviceArchitecture">
+    <Member Name="unknown" Value="0"/>
+    <Member Name="x86" Value="1"/>
+    <Member Name="x64" Value="2"/>
+    <Member Name="arm" Value="3"/>
+    <Member Name="arm64" Value="4"/>
+    <Member Name="unknownFutureValue" Value="5"/>
+</EnumType>
+```
+
+The enum was later extended to add a new value of ```quantum``` leading to the below CSDL
+
+```xml
+<EnumType Name="managedDeviceArchitecture">
+    <Member Name="unknown" Value="0"/>
+    <Member Name="x86" Value="1"/>
+    <Member Name="x64" Value="2"/>
+    <Member Name="arm" Value="3"/>
+    <Member Name="arm64" Value="4"/>
+    <Member Name="unknownFutureValue" Value="5"/>
+    <Member Name="quantum" Value="6"/>
+</EnumType>
+```
+
+### Default Behavior
+
+```http
+GET https://graph.microsoft.com/v1.0/deviceManagement/managedDevices?$select=displayName,processorArchitecture
+```
+
+```json
+{
+    "value": [
+        { 
+            "id": "0", 
+            "displayName": "Surface Pro X", 
+            "processorArchitecture" : "arm64"
+        },
+        { 
+            "id": "1",
+            "displayName": "Prototype",
+            "processorArchitecture": "unknownFutureValue"
+        }
+        { 
+            "id": "2",
+            "displayName": "My Laptop", 
+            "processorArchitecture": "x64"
+        }
+    ]
+}
+```
+
+In this case the value of the ```processorArchitecture``` property is ```quantum``` however since the client did not request the ```include-unknown-enum-members``` header the value was replaced with ```unknownFutureValue```
+
+### Include opt-in Header
+
+```http
+GET https://graph.microsoft.com/v1.0/deviceManagement/managedDevices?$select=displayName,processorArchitecture
+
+Preference: include-unknown-enum-members
+```
+
+```json
+{
```

Comment:
Missing the Preference-Applied header.  Same for the last example in the doc.

Reply:
```suggestion
Preference-Applied: include-unknown-enum-members

{
```